include/seqan3/io/sequence_file/format_sam.hpp

```diff
@@ -0,0 +1,308 @@
+              typename seq_legal_alph_type, bool seq_qual_combined,
+              typename seq_type,        // other constraints checked inside function
+              typename id_type,
+              typename qual_type>
+    void read(stream_type                     & stream,
+              sequence_file_input_options<seq_legal_alph_type, seq_qual_combined> const & options,
+              seq_type                        & sequence,
+              id_type                         & id,
+              qual_type                       & qualities)
+    {
+        auto stream_view = view::subrange<decltype(std::istreambuf_iterator<char>{stream}),
+                                          decltype(std::istreambuf_iterator<char>{})>
+                            {std::istreambuf_iterator<char>{stream},
+                             std::istreambuf_iterator<char>{}};
+
+        // cache the begin position so we write quals to the same position as seq in seq_qual case
+        size_t sequence_size_before, sequence_size_after = 0;
+        if constexpr (!detail::decays_to_ignore_v<seq_type>)
+            sequence_size_before = ranges::size(sequence);
+
+        auto constexpr is_tab = is_char<'\t'>;
+        auto constexpr is_cntrl= is_char<'\n'>;
+
+        while (is_char<'@'>(*ranges::begin(stream_view)))
+            detail::consume(stream_view | view::take_line);
+
+        /* ID */
+        std::string first; // Checking if id is given or set to *
+        ranges::copy(stream_view | view::take_until_or_throw(is_tab || is_blank), std::back_inserter(first));
+        if (first == "*")
+            throw parse_error{"The ID field may not be empty for SAM files."};
+
+        if constexpr (!detail::decays_to_ignore_v<id_type>)
+        {
+            ranges::copy(first | view::char_to<value_type_t<id_type>>, std::back_inserter(id)); //Add first to id
+
+            if (options.truncate_ids)
+            {
+              detail::consume(stream_view | view::take_until_or_throw(is_tab));
+            }
+            else
+            {
+              ranges::copy(stream_view | view::take_until_or_throw(is_tab)
+                                       | view::char_to<value_type_t<id_type>>,
+                           std::back_inserter(id));
+            }
+        }
+        else
+        {
+             detail::consume(stream_view | view::take_until_or_throw(is_tab));
+        }
+
+        //Jump over SAM colums 2-9 (FLAG RNAME POS MAPQ CIGAR RNEXT PNEXT TLEN)
+        for (int i = 0; i < 8;i++)
+        {
+            ++begin(stream_view);
+            detail::consume(stream_view | view::take_until_or_throw(is_tab));
+        }
+        ++begin(stream_view);
+
+        auto seq_view = stream_view | view::take_until_or_throw(is_tab); // until next tab
+
+         // Sequence
+         if constexpr (!detail::decays_to_ignore_v<seq_type>)
+         {
+             auto constexpr is_legal_alph = is_in_alphabet<seq_legal_alph_type>;
```

Comment:
don't you also need to check if the sequence is `*`?

Reply:
At the moment it will throw an parse_error{std::string{"Encountered an unexpected letter: "}. Do you think this error message is misleading? Then I will add another if before.

Reply:
Yes for consistency :)